### ISSUE TITLE
API/BUG: type of scalar aggregations

### ISSUE THREAD

**Issue by chris-b1** [pandas-dev/pandas#15385]
Currently there is some inconsistency around the scalar type returned from a `Series` aggregation, both in terms of whether it is a numpy or python type, as well as different behavior for an empty `Series` - see table below.

Normally this isn't a big deal as the numpy types mostly behave like the python type, but can be an issue with serialization, which is where I ran into this.

Is the desired behavior to make all of these python types?

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>function</th>
      <th>type</th>
      <th>type_empty</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>sum</td>
      <td>&lt;class 'float'&gt;</td>
      <td>&lt;class 'int'&gt;</td>
    </tr>
    <tr>
      <td>mean</td>
      <td>&lt;class 'float'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>median</td>
      <td>&lt;class 'float'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>count</td>
      <td>&lt;class 'numpy.int32'&gt;</td>
      <td>&lt;class 'numpy.int32'&gt;</td>
    </tr>
    <tr>
      <td>var</td>
      <td>&lt;class 'float'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>std</td>
      <td>&lt;class 'float'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>sem</td>
      <td>&lt;class 'numpy.float64'&gt;</td>
      <td>&lt;class 'numpy.float64'&gt;</td>
    </tr>
    <tr>
      <td>nunique</td>
      <td>&lt;class 'int'&gt;</td>
      <td>&lt;class 'int'&gt;</td>
    </tr>
    <tr>
      <td>prod</td>
      <td>&lt;class 'numpy.float64'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>min</td>
      <td>&lt;class 'numpy.float64'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
    <tr>
      <td>max</td>
      <td>&lt;class 'numpy.float64'&gt;</td>
      <td>&lt;class 'float'&gt;</td>
    </tr>
  </tbody>
</table>

Code for table
```python
fxs = ['sum', 'mean', 'median', 'count', 'var', 'std', 'sem',  'nunique', 'prod', 'min', 'max']
s = pd.Series([1., 2., 3.])
s_empty = pd.Series([], dtype='f8')
data = []

for f in fxs:
    row = dict(function=f)
    res = getattr(s, f)()
    row['type'] = type(res)
    res = getattr(s_empty, f)()
    row['type_empty'] = type(res)
    data.append(row)

pd.DataFrame(data).to_html(index=False)
```

**Comment by jreback** [comment]
yes this would be good to make consistent. scalars should always be python scalars; numpy scalars are a weird hybrid that can have odd effects.

**Comment by chris-b1** [comment]
Extra test case from #19381

```python
import pandas as pd
import json
import datetime

data = [
    datetime.date(1987, 2, 12),
    datetime.date(1987, 2, 12),
    datetime.date(1987, 2, 12),
    None,
    None,
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15),
    datetime.date(1989, 6, 15)
]

df = pd.DataFrame(columns=['foo'])
df['foo'] = data
ds = df['foo'].describe()
d = ds.to_dict()
j = json.dumps(d)
print(j)
```

**Comment by torlenor** [comment]
This is still an issue and leads to troubles, as the original poster described, in serialization when packages are used which do not know how to handle numpy types.

**Comment by jreback** [comment]
@torlenor pull requests for tests and patches are always welcome and how things get fixed

**Comment by jbrockmendel** [comment]
@rhshadrach possibly addressed by your python-types work?

**Comment by rhshadrach** [comment]
Closed by #63016
